core/src/main/java/org/infinispan/container/versioning/RandomVersionGenerator.java

```diff
@@ -0,0 +1,34 @@
+package org.infinispan.container.versioning;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ *
+ */
+public final class RandomVersionGenerator implements VersionGenerator {
+
+   public static final RandomVersionGenerator INSTANCE = new RandomVersionGenerator();
+
+   private RandomVersionGenerator() {}
+
+   @Override
+   public NumericVersion generateNew() {
+      // Hot Rod streaming cache expects a version higher than zero.
+      long version;
+      do {
+         version = Math.abs(ThreadLocalRandom.current().nextLong());
+      } while (version <= 0);
```

Comment:
Looks like it can be removed.

Reply:
ops... I added the `Math.abs` and forgot to remove the loop :facepalm: